Firestore/core/src/api/aggregate_query.cc

```diff
@@ -42,11 +43,28 @@ AggregateQuery::AggregateQuery(Query query,
     : query_{std::move(query)}, aggregates_{std::move(aggregates)} {
 }
 
-void AggregateQuery::Get(AggregateQueryCallback&& callback) {
+void AggregateQuery::GetAggregate(AggregateQueryCallback&& callback) {
   query_.firestore()->client()->RunAggregateQuery(query_.query(), aggregates_,
                                                   std::move(callback));
 }
 
+// TODO(b/280805906) Remove this count specific API after the c++ SDK migrates
+// to the new Aggregate API
+void AggregateQuery::Get(CountQueryCallback&& callback) {
+  this->GetAggregate([callback](const StatusOr<ObjectValue>& result) {
+    if (result.ok()) {
+      auto count_value =
+          result.ValueOrDie().Get(AggregateAlias("count").StringValue());
+      HARD_ASSERT(count_value.has_value() &&
+                  count_value.value().which_value_type ==
+                      google_firestore_v1_Value_integer_value_tag);
+      callback(StatusOr<int64_t>(count_value->integer_value));
+    } else {
+      callback(StatusOr<int64_t>(std::move(result.status())));
```

Comment:
Optional: Consider changing the structure of this method to reduce the amount of nesting, to improve readability. I'd recommend to change `if (result.ok())` to `if (!result.ok())` and in its block call the callback and return. The rest of the function's body can then be un-indented, since that is the "typical" path.

Reply:
I'm not a fan of that pattern when it requires an additional `return` statement because the code becomes more fragile.  A missed return statement could still show expected behavior but fall through and execute unexpected code. In a scenario not requiring the extra return, I agree.

Reply:
modified this. After reading a few of the linked articles, I see the point of view.